api/lib/users_functions.php

```diff
@@ -25,19 +25,19 @@
 
 /* Return the groups of a user */
 function user_membership($user) {
-    $output = shell_exec("/usr/libexec/nethserver/list-user-membership -s '$user'");
+    $output = shell_exec(escapeshellcmd("/usr/libexec/nethserver/list-user-membership -s ".escapeshellarg($user)));
```

Comment:
`escapeshellcmd()` should not be needed because the inner `escapeshellarg()` does all the job.

```suggestion
    $output = shell_exec("/usr/libexec/nethserver/list-user-membership -s " . escapeshellarg($user));
```

I'd call `escapeshellcmd()` only on a whole shell command line where individual arguments are not under my control. That's not what happens here: our `$user` variable corresponds to an argument and is the only argument to protect.